raster/src/main/scala/geotrellis/raster/FeatureExtraction.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.raster.rasterize.Rasterizer
+import geotrellis.vector.{Feature, Geometry, Point, PointFeature}
+
+import spire.syntax.cfor._
+
+import scala.collection.mutable.ListBuffer
+
+trait FeatureExtraction[I <: Geometry, T <: CellGrid[Int], O <: Geometry, D] {
+  def features(geom: I, raster: Raster[T]): Array[Array[Feature[O, D]]]
```

Comment:
@pomadchin asked for input and we discussed changing this return type to `Array[Feature[O, Array[D]]]` so that we don't duplicate point data and its clearer that the data associated with the feature is the int or double value for each band in the queried raster. He's going to attempt this change. At a glance, we don't think that this makes any particular downstream operation against the result data any more difficult than it would have been with the old type `Array[Array[Feature[O, D]]]`

Reply:
:+1: